components/accelerator/nvidia/fabric-manager/component.go

```diff
@@ -164,10 +134,34 @@ func (c *component) Metrics(ctx context.Context, since time.Time) ([]components.
 
 func (c *component) Close() error {
 	log.Logger.Debugw("closing component")
+	c.cancel()
 
-	// safe to call stop multiple times
-	_ = c.poller.Stop(Name)
-	c.logPoller.Stop(Name)
+	if c.logLineProcessor != nil {
+		c.logLineProcessor.close()
+	}
+	if c.eventsStore != nil {
+		c.eventsStore.Close()
+	}
 
 	return nil
 }
+
+func fabricManagerExists() bool {
+	p, err := exec.LookPath("nv-fabricmanager")
+	if err != nil {
+		return false
+	}
+	return p != ""
+}
+
+func checkFabricManagerActive(ctx context.Context, conn *pkg_systemd.DbusConn) (bool, error) {
+	active, err := conn.IsActive(ctx, "nvidia-fabricmanager")
+	if err != nil {
+		return false, err
+	}
+	return active, nil
+}
+
+func getLatestFabricManagerOutput(ctx context.Context) (string, error) {
```

Comment:
BTW, we can deprecate the function, just  directly call `pkg_systemd.GetLatestJournalctlOutput`..